osfoffline/polling_osf_manager/osf_query.py

```diff
@@ -132,7 +135,13 @@ def upload_file(self, local_file):
         files_url = api_url_for(RESOURCES, node_id=local_file.node.osf_id, provider=local_file.provider,
                                 file_id=parent_osf_id)
         file = open(local_file.path, 'rb')
-        resp_json = yield from self.make_request(files_url, method="PUT", params=params, data=file, get_json=True)
+
+        file_lock = asyncio.Lock()
+        file_data = safe_file_stream(file, file_lock)
+        file_size = os.path.getsize(local_file.path)
+
+        resp_json = yield from self.make_request(files_url, method="PUT", params=params, data=file_data, get_json=True,
+            headers={'Content-Length': str(file_size)}, local_file=local_file, locks=[file_lock, asyncio.Lock()])
```

Comment:
what if there is a http error or server is down during this request. There is no exception being raised here.